cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/QueryPlanner.scala

```diff
@@ -285,4 +285,103 @@ object QueryPlanner {
     JsArray(jsArray.value.map(obj => JsObject(obj.asInstanceOf[JsObject].fields.filter(e => project.contains(e._1)))))
   }
 
+  // Handle avg function for ProgressiveSolver
+  def handleAvg(mergedResults:Seq[JsArray]):Seq[JsArray] =
```

Comment:
These two function looks so identical. Can you refactor these two come up with more compact functions? Also, can you change the comments of these two functions to let us know that which function is used when?

Reply:
Solved.